infra/modules/providers/azure/storage-account/README.md

```diff
@@ -0,0 +1,96 @@
+
+- For authentication, this module automatically enrolls the deployed `storage account` into Microsoft's _**Managed Identities**_ service.
+
+- For authorization, this module expects to receive the `object id` of any Azure Resource that will behave as a client of the `storage-account`. Enrollment as an Azure Resource client is achieved through Microsoft's _**Role Based Access Secruity**_ (RBAC) service.
+
+#### _More on Azure Storage_
+
+> "A storage account provides a unique namespace in Azure for your data. Every object that you store in Azure Storage has an address that includes your unique account name. A container organizes a set of blobs, similar to a directory in a file system. A storage account can include an unlimited number of containers, and a container can store an unlimited number of blobs." - Source: Microsoft's [Introduction to Azure Blob Storage](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction)
+
+This module deploys a `storage account` along with a `storage container` in order to satisfy blob storage scenarios which are optimized for storing massive amounts of unstructured data, such as text or binary data.
+
+## Characteristics
+
+An instance of the `storage-account` module deploys the _**Azure Storage**_ service in order to provide templates with the following:
+
+- Ability to deploy a Storage Account automatically enrolled into _**Managed Identities**_ within a single resource group.
+
+- Ability to deploy a Storage Account with a role assigned Azure Resource client.
+
+- Ability to deploy a Storage Container within the Storage Account.
+
+## Definition
+
+Terraform resources directly referenced within the `storage-account` module include the following:
+
+- [azurerm_storage_account](https://www.terraform.io/docs/providers/azurerm/r/storage_account.html)
+
+- [azurerm_storage_container](https://www.terraform.io/docs/providers/azurerm/r/storage_container.html)
+
+## Usage
+
+Azure Storage usage example:
+
+```terraform
+resource "azurerm_resource_group" "main" {
+  name     = var.prefix
+  location = var.resource_group_location
+}
+
+module "service_plan" {
+  source              = "../../modules/providers/azure/service-plan"
+  resource_group_name = azurerm_resource_group.main.name
+  service_plan_name   = "${azurerm_resource_group.main.name}-sp"
+}
+
+module "app_service" {
+  source              = "../../modules/providers/azure/app-service"
+  app_service_name    = var.app_service_name
+  service_plan_name   = module.service_plan.service_plan_name
+  ...
+}
+
+module "storage_account" {
+  source                    = "../../modules/providers/azure/storage-account"
+  resource_group_name       = azurerm_resource_group.main.name
+  resource_group_location   = azurerm_resource_group.main.location
+  account_name              = "mystorageaccount"
+  storage_container_names   = ["astoragcontainer", "astoragecontainer"]
```

Comment:
Why would I specify 2 containers with the same name?

Reply:
Container names have been changed to be more example friendly. [ "movies", "photos"].